src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1900,7 +1900,7 @@ private void CheckMemberNameConflicts(BindingDiagnosticBag diagnostics)
                             if (symbol.Kind != SymbolKind.Field || !symbol.IsImplicitlyDeclared)
                             {
                                 // The type '{0}' already contains a definition for '{1}'
-                                if (Locations.Length == 1 || IsPartial)
+                                if ((Locations.Length == 1 || IsPartial) && symbol is not SourceIncompleteMemberFieldSymbol)
```

Comment:
>&& symbol is not SourceIncompleteMemberFieldSymbol [](http://example.com/codeflow?start=73&length=50)

I am not sure if this special case is worth having. Does it cover some common scenario?

Reply:
Without it, we'll produce unnecessarily extra (and confusing) diagnostics.

Reply:
> Without it, we'll produce unnecessarily extra (and possibly confusing) diagnostics.

For the purpose of review, it would be useful to see examples for some scenarios

Reply:
I think I'll need to add a similar check. This is for the tests which have extra `ERR_DuplicateNameInClass` errors now.

Reply:
@AlekseyTs We can also consider not creating a symbol for such incomplete declarations. What do you think?

Reply:
> We can also consider not creating a symbol for such incomplete declarations. What do you think?

What the impact of that is going to be? What else is going to be affected and how? I assume, GetDeclaredSymbol API stops returning symbols. I thought the goal was exactly the opposite. Quite possibly other SemanticModel behaviors around the syntax are going to be affected. But, without knowing how exactly, it is hard to say whether that would be acceptable. 

It feels strange to care about the name length only in this particular place of this function. Could we move the length check above, before we enter ```foreach (var symbol in pair.Value)```? If ```name``` is empty, then there is a syntax error and we can skip analysis for the entire group.

Reply:
> I assume, GetDeclaredSymbol API stops returning symbols.

@AlekseyTs Isn't it the opposite?

https://github.com/dotnet/roslyn/blob/eebc8b739b34875bd3a2f6c28a3fe2c327ca6405/src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs#L1443-L1467

Reply:
Here is a comment there:
```
// these declarations can contain multiple variable declarators. GetDeclaredSymbol should be called on them directly.
```

A declarator should be used to get the symbol.